jdaviz/configs/imviz/plugins/tools.py

```diff
@@ -134,59 +134,56 @@ class ContrastBias(CheckableTool):
     tool_tip = 'Click and drag to adjust contrast and bias, double-click to reset'
 
     def __init__(self, viewer, **kwargs):
-        self._time_last = 0
         super().__init__(viewer, **kwargs)
 
     def activate(self):
         self.viewer.add_event_callback(self.on_mouse_or_key_event,
                                        events=['dragstart', 'dragmove',
-                                               'dragend', 'dblclick'])
+                                               'dblclick'])
 
     def deactivate(self):
         self.viewer.remove_event_callback(self.on_mouse_or_key_event)
 
-    def on_mouse_or_key_event(self, data):
-        from jdaviz.configs.imviz.helper import get_top_layer_index
+    @debounced(delay_seconds=0.03, method=True)
+    def _dragevent(self, data):
+        event_x = data['pixel']['x']
+        event_y = data['pixel']['y']
 
-        event = data['event']
+        if ((event_x < 0) or (event_x >= self._max_x) or
+                (event_y < 0) or (event_y >= self._max_y)):
+            return
 
-        # Note that we throttle this to 200ms here as changing the contrast
-        # and bias it expensive since it forces the whole image to be redrawn
-        if event == 'dragmove':
-            if (time.time() - self._time_last) <= 0.2:
-                return
+        # Normalize w.r.t. viewer display from 0 to 1
+        # Also see glue/viewers/matplotlib/qt/toolbar_mode.py
+        # bias range 0..1
+        # contrast range 0..4
+        bias = event_x / (self._max_x - 1)
```

Comment:
Huh, you know, I always thought this worked based on relative position to where you initially clicked, not compared to the edges of the viewer. No wonder I had a hard time using it 😆

Reply:
Kyle did change the math to relative calculation but I made him change it back for this PR so we can compare performance on the mouse drag without worrying about the math.

If you think absolute calculation is the wrong way to do this, we can revisit in a future PR. Though with relative math, I am not sure how one can tell the mouse to drag to min or max in a consistent way. 🤷